<h1>Project 5 – What would it take to Meet the Paris Agreement Energy Production Targets?<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><ul class="toc-item"><li><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Energy-Demand-Projection" data-toc-modified-id="Energy-Demand-Projection-0.0.0.1"><span class="toc-item-num">0.0.0.1&nbsp;&nbsp;</span>Energy Demand Projection</a></span></li><li><span><a href="#Renaming-df_projected_demand" data-toc-modified-id="Renaming-df_projected_demand-0.0.0.2"><span class="toc-item-num">0.0.0.2&nbsp;&nbsp;</span>Renaming df_projected_demand</a></span></li></ul></li><li><span><a href="#Visualize-the-quantities-of-the-data." data-toc-modified-id="Visualize-the-quantities-of-the-data.-0.0.1"><span class="toc-item-num">0.0.1&nbsp;&nbsp;</span>Visualize the quantities of the data.</a></span></li></ul></li><li><span><a href="#Preparing-Data-For-Analysis" data-toc-modified-id="Preparing-Data-For-Analysis-0.1"><span class="toc-item-num">0.1&nbsp;&nbsp;</span>Preparing Data For Analysis</a></span><ul class="toc-item"><li><span><a href="#Testing-for-bad-data" data-toc-modified-id="Testing-for-bad-data-0.1.1"><span class="toc-item-num">0.1.1&nbsp;&nbsp;</span>Testing for bad data</a></span></li></ul></li><li><span><a href="#Frequency-Distributions" data-toc-modified-id="Frequency-Distributions-0.2"><span class="toc-item-num">0.2&nbsp;&nbsp;</span>Frequency Distributions</a></span></li></ul></li><li><span><a href="#Building-Models" data-toc-modified-id="Building-Models-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Building Models</a></span><ul class="toc-item"><li><span><a href="#Final-Model-Selection" data-toc-modified-id="Final-Model-Selection-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Final Model Selection</a></span></li></ul></li><li><span><a href="#Tuning-Models" data-toc-modified-id="Tuning-Models-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Tuning Models</a></span></li><li><span><a href="#Final-Model-And-Analysis" data-toc-modified-id="Final-Model-And-Analysis-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Final Model And Analysis</a></span></li><li><span><a href="#Conclusion" data-toc-modified-id="Conclusion-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Conclusion</a></span><ul class="toc-item"><li><span><a href="#Limitations-of-Scope-/-Provided-Data" data-toc-modified-id="Limitations-of-Scope-/-Provided-Data-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Limitations of Scope / Provided Data</a></span></li><li><span><a href="#Future-Research-Opportunities" data-toc-modified-id="Future-Research-Opportunities-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>Future Research Opportunities</a></span></li><li><span><a href="#Recommendations" data-toc-modified-id="Recommendations-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Recommendations</a></span></li><li><span><a href="#Grading" data-toc-modified-id="Grading-4.4"><span class="toc-item-num">4.4&nbsp;&nbsp;</span>Grading</a></span><ul class="toc-item"><li><span><a href="#Attention-to-Detail" data-toc-modified-id="Attention-to-Detail-4.4.1"><span class="toc-item-num">4.4.1&nbsp;&nbsp;</span>Attention to Detail</a></span><ul class="toc-item"><li><span><a href="#Exceeds-Objective" data-toc-modified-id="Exceeds-Objective-4.4.1.1"><span class="toc-item-num">4.4.1.1&nbsp;&nbsp;</span>Exceeds Objective</a></span></li><li><span><a href="#Meets-Objective-(Passing-Bar)" data-toc-modified-id="Meets-Objective-(Passing-Bar)-4.4.1.2"><span class="toc-item-num">4.4.1.2&nbsp;&nbsp;</span>Meets Objective (Passing Bar)</a></span></li></ul></li><li><span><a href="#Project-Motivation" data-toc-modified-id="Project-Motivation-4.4.2"><span class="toc-item-num">4.4.2&nbsp;&nbsp;</span>Project Motivation</a></span><ul class="toc-item"><li><span><a href="#Exceeds-Objective" data-toc-modified-id="Exceeds-Objective-4.4.2.1"><span class="toc-item-num">4.4.2.1&nbsp;&nbsp;</span>Exceeds Objective</a></span></li><li><span><a href="#Meets-Objective-(Passing-Bar)" data-toc-modified-id="Meets-Objective-(Passing-Bar)-4.4.2.2"><span class="toc-item-num">4.4.2.2&nbsp;&nbsp;</span>Meets Objective (Passing Bar)</a></span></li></ul></li><li><span><a href="#Independent-Learning" data-toc-modified-id="Independent-Learning-4.4.3"><span class="toc-item-num">4.4.3&nbsp;&nbsp;</span>Independent Learning</a></span><ul class="toc-item"><li><span><a href="#Exceeds-Objective" data-toc-modified-id="Exceeds-Objective-4.4.3.1"><span class="toc-item-num">4.4.3.1&nbsp;&nbsp;</span>Exceeds Objective</a></span></li><li><span><a href="#Meets-Objective-(Passing-Bar)" data-toc-modified-id="Meets-Objective-(Passing-Bar)-4.4.3.2"><span class="toc-item-num">4.4.3.2&nbsp;&nbsp;</span>Meets Objective (Passing Bar)</a></span></li></ul></li><li><span><a href="#Reproducibility" data-toc-modified-id="Reproducibility-4.4.4"><span class="toc-item-num">4.4.4&nbsp;&nbsp;</span>Reproducibility</a></span><ul class="toc-item"><li><span><a href="#Exceeds-Objective" data-toc-modified-id="Exceeds-Objective-4.4.4.1"><span class="toc-item-num">4.4.4.1&nbsp;&nbsp;</span>Exceeds Objective</a></span></li><li><span><a href="#Meets-Objective-(Passing-Bar)" data-toc-modified-id="Meets-Objective-(Passing-Bar)-4.4.4.2"><span class="toc-item-num">4.4.4.2&nbsp;&nbsp;</span>Meets Objective (Passing Bar)</a></span></li></ul></li><li><span><a href="#Legacy-Data-Sources" data-toc-modified-id="Legacy-Data-Sources-4.4.5"><span class="toc-item-num">4.4.5&nbsp;&nbsp;</span>Legacy Data Sources</a></span><ul class="toc-item"><li><span><a href="#Disaster-Stats" data-toc-modified-id="Disaster-Stats-4.4.5.1"><span class="toc-item-num">4.4.5.1&nbsp;&nbsp;</span>Disaster Stats</a></span></li><li><span><a href="#Refugee-Stats" data-toc-modified-id="Refugee-Stats-4.4.5.2"><span class="toc-item-num">4.4.5.2&nbsp;&nbsp;</span>Refugee Stats</a></span></li><li><span><a href="#Population-Stats" data-toc-modified-id="Population-Stats-4.4.5.3"><span class="toc-item-num">4.4.5.3&nbsp;&nbsp;</span>Population Stats</a></span></li><li><span><a href="#GDP-Stats" data-toc-modified-id="GDP-Stats-4.4.5.4"><span class="toc-item-num">4.4.5.4&nbsp;&nbsp;</span>GDP Stats</a></span></li><li><span><a href="#At-Risk-Property-Stats" data-toc-modified-id="At-Risk-Property-Stats-4.4.5.5"><span class="toc-item-num">4.4.5.5&nbsp;&nbsp;</span>At-Risk Property Stats</a></span></li><li><span><a href="#Food-Production" data-toc-modified-id="Food-Production-4.4.5.6"><span class="toc-item-num">4.4.5.6&nbsp;&nbsp;</span>Food Production</a></span></li><li><span><a href="#The-2030-target,-keeping-with-the-Paris-Agreement" data-toc-modified-id="The-2030-target,-keeping-with-the-Paris-Agreement-4.4.5.7"><span class="toc-item-num">4.4.5.7&nbsp;&nbsp;</span>The 2030 target, keeping with the Paris Agreement</a></span></li></ul></li></ul></li></ul></li><li><span><a href="#Possible-Project-Directions:" data-toc-modified-id="Possible-Project-Directions:-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Possible Project Directions:</a></span><ul class="toc-item"><li><span><a href="#Show-Potential-Donors-That-The-Financial-Cost-of-Disaster-Mitigation-Efforts-Will-Continue-To-Rise" data-toc-modified-id="Show-Potential-Donors-That-The-Financial-Cost-of-Disaster-Mitigation-Efforts-Will-Continue-To-Rise-5.1"><span class="toc-item-num">5.1&nbsp;&nbsp;</span>Show Potential Donors That The Financial Cost of Disaster Mitigation Efforts Will Continue To Rise</a></span><ul class="toc-item"><li><span><a href="#Stakeholder:-FEMA?-UN-Disaster-Org?-Red-Cross?" data-toc-modified-id="Stakeholder:-FEMA?-UN-Disaster-Org?-Red-Cross?-5.1.1"><span class="toc-item-num">5.1.1&nbsp;&nbsp;</span>Stakeholder: FEMA? UN Disaster Org? Red Cross?</a></span></li></ul></li><li><span><a href="#Illustrate-Climate-Change's-Future-Effects-on-the-Global-Food-Supply" data-toc-modified-id="Illustrate-Climate-Change's-Future-Effects-on-the-Global-Food-Supply-5.2"><span class="toc-item-num">5.2&nbsp;&nbsp;</span>Illustrate Climate Change's Future Effects on the Global Food Supply</a></span><ul class="toc-item"><li><span><a href="#Stakeholder:-Hunger-NGO?-Compassion-International?-Food-For-The-Hungry?" data-toc-modified-id="Stakeholder:-Hunger-NGO?-Compassion-International?-Food-For-The-Hungry?-5.2.1"><span class="toc-item-num">5.2.1&nbsp;&nbsp;</span>Stakeholder: Hunger NGO? Compassion International? Food For The Hungry?</a></span></li></ul></li><li><span><a href="#Refugee-projection-for-the-United-States" data-toc-modified-id="Refugee-projection-for-the-United-States-5.3"><span class="toc-item-num">5.3&nbsp;&nbsp;</span>Refugee projection for the United States</a></span><ul class="toc-item"><li><span><a href="#Stakeholder:-The-Subcommittee-on-Immigration-Integrity,-Security,-and-Enforcement" data-toc-modified-id="Stakeholder:-The-Subcommittee-on-Immigration-Integrity,-Security,-and-Enforcement-5.3.1"><span class="toc-item-num">5.3.1&nbsp;&nbsp;</span>Stakeholder: <a href="https://judiciary.house.gov/subcommittees/committee-judiciary/subcommittee-immigration-integrity-security-and-enforcement" target="_blank">The Subcommittee on Immigration Integrity, Security, and Enforcement</a></a></span></li></ul></li><li><span><a href="#The-SSP-quantifications-and-the-data-in-this-database-should-be-cited-as:" data-toc-modified-id="The-SSP-quantifications-and-the-data-in-this-database-should-be-cited-as:-5.4"><span class="toc-item-num">5.4&nbsp;&nbsp;</span>The SSP quantifications and the data in this database should be cited as:</a></span><ul class="toc-item"><li><span><a href="#In-addition,-please-cite-also-the-following-papers-if-you-are-using" data-toc-modified-id="In-addition,-please-cite-also-the-following-papers-if-you-are-using-5.4.1"><span class="toc-item-num">5.4.1&nbsp;&nbsp;</span>In addition, please cite also the following papers if you are using</a></span></li><li><span><a href="#For-the-five-SSP-IAM-&quot;marker&quot;-scenario-sets-please-use-the-following-citations:" data-toc-modified-id="For-the-five-SSP-IAM-&quot;marker&quot;-scenario-sets-please-use-the-following-citations:-5.4.2"><span class="toc-item-num">5.4.2&nbsp;&nbsp;</span>For the five SSP IAM "marker" scenario sets please use the following citations:</a></span></li><li><span><a href="#For-the-basic-elements-of-the-SSPs-please-cite-in-addition-to-the-SSP-overview-also-the-following-papers:" data-toc-modified-id="For-the-basic-elements-of-the-SSPs-please-cite-in-addition-to-the-SSP-overview-also-the-following-papers:-5.4.3"><span class="toc-item-num">5.4.3&nbsp;&nbsp;</span>For the basic elements of the SSPs please cite in addition to the SSP overview also the following papers:</a></span></li><li><span><a href="#Additional-detail-about-the-energy,-Land-use-and-air-pollution-trends-of-the-SSP-IAM-scenarios-can-be-found-here:" data-toc-modified-id="Additional-detail-about-the-energy,-Land-use-and-air-pollution-trends-of-the-SSP-IAM-scenarios-can-be-found-here:-5.4.4"><span class="toc-item-num">5.4.4&nbsp;&nbsp;</span>Additional detail about the energy, Land-use and air pollution trends of the SSP IAM scenarios can be found here:</a></span></li><li><span><a href="#Historical-Emissions-Inventories:" data-toc-modified-id="Historical-Emissions-Inventories:-5.4.5"><span class="toc-item-num">5.4.5&nbsp;&nbsp;</span>Historical Emissions Inventories:</a></span></li></ul></li></ul></li></ul></div>

#### Energy Demand Projection
* Energy demand projection until 2050: [IEA](https://www.iea.org/data-and-statistics/data-product/world-energy-outlook-2022-free-dataset)

In [11]:
df_projected_demand_raw = pd.read_csv(
    "Data/IEA-WEO2022_AnnexA_Free_Dataset_Regions.csv",
    encoding='ANSI')
df_projected_demand_raw

PUBLICATION                    SCENARIO  \
0     World Energy Outlook 2022    Stated Policies Scenario   
1     World Energy Outlook 2022    Stated Policies Scenario   
2     World Energy Outlook 2022    Stated Policies Scenario   
...                         ...                         ...   
2927  World Energy Outlook 2022    Stated Policies Scenario   
2928  World Energy Outlook 2022  Announced Pledges Scenario   
2929  World Energy Outlook 2022  Announced Pledges Scenario   

                        CATEGORY PRODUCT                     FLOW    UNIT  \
0                         Energy   Total      Total energy supply      PJ   
1                         Energy   Total      Total energy supply      PJ   
2                         Energy   Total      Total energy supply      PJ   
...                          ...     ...                      ...     ...   
2927  CO2 combustion and process   Total  Total final consumption  Mt CO2   
2928  CO2 combustion and process   Total  Total final consumption  Mt CO2   
2929  CO2 combustion and process   Total  Total final consumption  Mt CO2   

              REGION  YEAR     VALUE  
0              World  2010   541.981  
1              World  2020   592.311  
2              World  2021   624.164  
...              ...   ...       ...  
2927  Southeast Asia  2050  1422.280  
2928  Southeast Asia  2030  1081.410  
2929  Southeast Asia  2050   670.950  

[2930 rows x 9 columns]

In [24]:
df_projected_demand_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2930 entries, 0 to 2929
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PUBLICATION  2930 non-null   object 
 1   SCENARIO     2930 non-null   object 
 2   CATEGORY     2930 non-null   object 
 3   PRODUCT      2930 non-null   object 
 4   FLOW         2930 non-null   object 
 5   UNIT         2930 non-null   object 
 6   REGION       2930 non-null   object 
 7   YEAR         2930 non-null   int64  
 8   VALUE        2930 non-null   float64
dtypes: float64(1), int64(1), object(7)
memory usage: 206.1+ KB


#### Renaming df_projected_demand

In [41]:
# Renaming Columns
df_projected_demand = df_projected_demand_raw.copy()
renaming = {'PUBLICATION' : 'Publication',
            'SCENARIO' : 'Scenario',
            'CATEGORY' : 'Category',
            'PRODUCT' : 'Product',
            'FLOW' : 'Flow',
            'UNIT' : 'Unit',
            'REGION' : 'Region',
            'YEAR' : 'Year',
            'VALUE' : 'Value'}
df_projected_demand.rename(columns = renaming, inplace = True)

In [42]:
# I don't know what the carbon data is in this 
df_projected_demand = df_projected_demand.loc[
    df_projected_demand['Category'] == 'Energy']

# Each cell of Publication, and now Category, has the same value, I'll delete
# both columns.
df_projected_demand.drop(columns = ['Publication', 'Category'], inplace = True)

# I'm intrigued by the data in the Scenario column. There are three categories:
df_projected_demand['Scenario'].value_counts()

Stated Policies Scenario               1796
Announced Pledges Scenario              768
Net Zero Emissions by 2050 Scenario      30
Name: Scenario, dtype: int64

In [43]:
# The Announced Pledges Scenario could hold some interesting data, as well as
# the Net Zero Emissions by 2050 Scenario. Putting this data into the same
# Dataframe as the information I need will not be possible, so I'll split it
# into three different scenarios. 

df_proj_stat = df_projected_demand.loc[
    df_projected_demand['Scenario'] == 'Stated Policies Scenario'].copy()
df_proj_stat.drop(columns = ['Scenario'], inplace = True)

df_proj_ann = df_projected_demand.loc[
    df_projected_demand['Scenario'] == 'Announced Pledges Scenario'].copy()
df_proj_ann.drop(columns = ['Scenario'], inplace = True)

df_proj_zero = df_projected_demand.loc[
    df_projected_demand[
        'Scenario'] == 'Net Zero Emissions by 2050 Scenario'].copy()
df_proj_zero.drop(columns = ['Scenario'], inplace = True)

In [44]:
# Now I want to see what the Product column does.
df_proj_ann['Product'].value_counts()

Total                          252
Natural gas                     86
Coal                            84
Hydrogen                        64
Renewables                      64
Oil                             62
Solar PV                        32
Wind                            32
Nuclear                         32
Other                            4
Products from NGLs               2
Biofuels                         2
Tight oil                        2
CTL, GTL and additives           2
Conventional crude oil           2
Total oil                        2
Refinery market share            2
Fuel oil                         2
Natural gas liquids              2
Coking coal                      2
Diesel                           2
Naphtha                          2
Direct use of crude oil          2
LPG and ethane                   2
Refinery products                2
Extra-heavy oil and bitumen      2
Total liquids                    2
Conventional gas                 2
Kerosene            

I already have a reliable source for a breakdown of different region's energy production. I was primarily interested in this DataFrame because it shows what the increase in demand will be. The future of energy product is the variable I will adjust in this project. Therefore, I think I only need the information in the Total Energy needs. I'll drop the rest. The net zero pledges don't have a total energy category, so I won't need to drop anything there.

In [45]:
df_proj_stat = df_proj_stat.loc[df_proj_stat['Product'] == 'Total']
df_proj_stat.drop(columns = 'Product', inplace = True)

df_proj_ann = df_proj_ann.loc[df_proj_ann['Product'] == 'Total']
df_proj_ann.drop(columns = 'Product', inplace = True)

# Now I'll check the "Flow" column
df_proj_stat['Flow'].value_counts()

Total energy supply        80
Industry                   80
Total final consumption    80
Transport                  80
Buildings                  80
Electricity generation     80
Refining capacity          45
Refinery runs              45
Name: Flow, dtype: int64

Again, I really only wanted the Total energy supply information from this DataFrame. I looked up each definition of these terms in the report, and learned that only one of them is relevant to this project, defined below.

[Industry](https://iea.blob.core.windows.net/assets/830fe099-5530-48f2-a7c1-11f35d510983/WorldEnergyOutlook2022.pdf#page=499): The sector includes fuel used within the manufacturing and construction industries. Key industry branches include iron and steel, chemical and petrochemical, cement, aluminium, and pulp and paper. Use by industries for the transformation of energy into another form or for the production of fuels is excluded and reported separately under other energy sector. There is an exception for fuel transformation in blast furnaces and coke ovens, which are reported within iron and steel. Consumption of fuels for the transport of goods is reported as part of the transport sector, while consumption by off‐road vehicles is reported under industry.

[Total energy supply (TES)](https://iea.blob.core.windows.net/assets/830fe099-5530-48f2-a7c1-11f35d510983/WorldEnergyOutlook2022.pdf#page=497): Represents domestic demand only and is broken down into electricity and heat generation, other energy sector and total final consumption.

[Electricity generation](https://iea.blob.core.windows.net/assets/830fe099-5530-48f2-a7c1-11f35d510983/WorldEnergyOutlook2022.pdf#page=490): Defined as the total amount of electricity generated by power only or combined heat and power plants including generation required for own use. This is also referred to as gross generation.

[Buildings](https://iea.blob.core.windows.net/assets/830fe099-5530-48f2-a7c1-11f35d510983/WorldEnergyOutlook2022.pdf#page=488): The buildings sector includes energy used in residential and services buildings. Services buildings include commercial and institutional buildings and other non‐specified buildings. Building energy use includes space heating and cooling, water heating, lighting, appliances and cooking equipment.

[Total final consumption (TFC)](https://iea.blob.core.windows.net/assets/830fe099-5530-48f2-a7c1-11f35d510983/WorldEnergyOutlook2022.pdf#page=497): Is the sum of consumption by the various end‐use sectors. TFC is broken down into energy demand in the following sectors: industry (including manufacturing, mining, chemicals production, blast furnaces and coke ovens), transport, buildings (including residential and services) and other (including agriculture and other non‐energy use). It excludes international marine and aviation bunkers, except at world level where it is included in the transport sector.

[Transport](https://iea.blob.core.windows.net/assets/830fe099-5530-48f2-a7c1-11f35d510983/WorldEnergyOutlook2022.pdf#page=498): Fuels and electricity used in the transport of goods or people within the national territory irrespective of the economic sector within which the activity occurs. This includes fuel and electricity delivered to vehicles using public roads or for use in rail vehicles; fuel delivered to vessels for domestic navigation; fuel delivered to aircraft for domestic aviation; and energy consumed in the delivery of fuels through pipelines. Fuel delivered to international marine and aviation bunkers is presented only at the world level and is excluded from the transport sector at a domestic level. 

In [46]:
# I want to see what these numbers say for the "World" region.
df_proj_stat.loc[(df_proj_stat['Region'] == 'World') & 
                 (df_proj_stat['Year'] == 2021)]

Flow                     Unit Region  Year      Value
2         Total energy supply                       PJ  World  2021    624.164
698         Refining capacity  Million barrels per day  World  2021    101.200
703             Refinery runs  Million barrels per day  World  2021     77.900
1157   Electricity generation                      TWh  World  2021  28333.900
1941  Total final consumption                       PJ  World  2021    439.103
2053                 Industry                       PJ  World  2021    166.738
2165                Transport                       PJ  World  2021    113.433
2277                Buildings                       PJ  World  2021    132.436

Wow. These conversions don't make a lick of sense.

$ 28333.9 \ TWh \times \frac{3.6 PJ}{TwH} = 102002.04 \ PJ$

If these numbers are accurate, that means that in 2021, the world generated more electricity than the total final consumption. Since electricity generation is one of the sectors added to Total Final consumption, that doesn't make a lot of sense.

I opened the report to see if I could confirm what was going on. Fortunately, I was able confirm that this was a mere typo. [Table 5.1 on Page 239](https://iea.blob.core.windows.net/assets/830fe099-5530-48f2-a7c1-11f35d510983/WorldEnergyOutlook2022.pdf#page=239) of the report shared these same numbers, but confirmed they are not in petajoules, but in exajoules, which equal a thousand petajoules. 

$ 28333.9 \ TWh \times \frac{.0036 EJ}{TwH} = 102 \ EJ$

I'll change the units on this.

In [116]:
for i in df_proj_stat.index:
    if df_proj_stat.loc[i,'Unit'] == 'PJ':
        df_proj_stat.loc[i,'Unit'] = 'EJ'
        
for i in df_proj_ann.index:
    if df_proj_ann.loc[i,'Unit'] == 'PJ':
        df_proj_ann.loc[i,'Unit'] = 'EJ'

for i in df_proj_zero.index:
    if df_proj_zero.loc[i,'Unit'] == 'PJ':
        df_proj_zero.loc[i,'Unit'] = 'EJ'
        
df_proj_stat.loc[(df_proj_stat['Region'] == 'World') & 
                 (df_proj_stat['Year'] == 2021)]

Unit Region  Year    Value
1155  TWh  World  2010  21538.9

I'm still not sure which of these values are relevant for this analysis. I'll compare the above numbers to the electricity generation database from BP and see how they compare. Fortunatley, the BP database has a value for the whole world in 2021, split across different electricity genertaion methods. If I add them up, I'll have a number to compare.

In [117]:
electricity_2021 = df_electricity.loc[(df_electricity['Country'] == 'World') &
                   (df_electricity['Year'] == 2021)].copy()
Twh = electricity_2021.iloc[:,3:].values
BP_2021 = np.sum(Twh)
BP_2021

20938.54

In [119]:
28333.900 / BP_2010

1.0286724862382954

I'd say a 1.9% difference means I'm on the right track. I'll use the electricity generation values from the IEA database.

In [50]:
# I'll now single out Electricity generation in Flow, and drop the column.

df_proj_stat = df_proj_stat.loc[
    df_proj_stat['Flow'] == 'Electricity generation']
df_proj_stat.drop(columns = 'Flow', inplace = True)

df_proj_ann = df_proj_ann.loc[
    df_proj_ann['Flow'] == 'Electricity generation']
df_proj_ann.drop(columns = 'Flow', inplace = True)

In [51]:
df_proj_stat

Unit                     Region  Year     Value
1155  TWh                      World  2010  21538.90
1156  TWh                      World  2020  26707.70
1157  TWh                      World  2021  28333.90
1158  TWh                      World  2030  34833.60
1159  TWh                      World  2050  49844.90
1162  TWh              North America  2010   5232.84
1163  TWh              North America  2020   5205.30
1164  TWh              North America  2021   5356.51
1165  TWh              North America  2030   5771.40
1166  TWh              North America  2050   7815.80
1169  TWh              United States  2010   4354.36
1170  TWh              United States  2020   4238.97
1171  TWh              United States  2021   4371.41
1172  TWh              United States  2030   4624.97
1173  TWh              United States  2050   6270.34
1176  TWh  Central and South America  2010   1130.13
1177  TWh  Central and South America  2020   1275.88
1178  TWh  Central and South America  2021   1331.03
1179  TWh  Central and South America  2030   1605.23
1180  TWh  Central and South America  2050   2591.74
1183  TWh                     Brazil  2010    515.71
1184  TWh                     Brazil  2020    621.20
1185  TWh                     Brazil  2021    638.70
1186  TWh                     Brazil  2030    762.49
1187  TWh                     Brazil  2050   1173.86
1190  TWh                     Europe  2010   4120.40
1191  TWh                     Europe  2020   3955.91
1192  TWh                     Europe  2021   4181.82
1193  TWh                     Europe  2030   4691.24
1194  TWh                     Europe  2050   5703.31
1197  TWh             European Union  2010   2956.10
1198  TWh             European Union  2020   2757.72
1199  TWh             European Union  2021   2963.31
1200  TWh             European Union  2030   3238.30
1201  TWh             European Union  2050   3689.06
1204  TWh                     Africa  2010    687.12
1205  TWh                     Africa  2020    835.03
1206  TWh                     Africa  2021    868.76
1207  TWh                     Africa  2030   1204.15
1208  TWh                     Africa  2050   2337.29
1211  TWh                Middle East  2010    829.42
1212  TWh                Middle East  2020   1202.75
1213  TWh                Middle East  2021   1232.84
1214  TWh                Middle East  2030   1651.34
1215  TWh                Middle East  2050   2886.16
1218  TWh                    Eurasia  2010   1251.26
1219  TWh                    Eurasia  2020   1366.89
1220  TWh                    Eurasia  2021   1455.00
1221  TWh                    Eurasia  2030   1539.56
1222  TWh                    Eurasia  2050   1937.40
1225  TWh                     Russia  2010   1036.12
1226  TWh                     Russia  2020   1086.62
1227  TWh                     Russia  2021   1158.08
1228  TWh                     Russia  2030   1176.51
1229  TWh                     Russia  2050   1376.48
1232  TWh               Asia Pacific  2010   8287.77
1233  TWh               Asia Pacific  2020  12866.00
1234  TWh               Asia Pacific  2021  13907.90
1235  TWh               Asia Pacific  2030  18370.70
1236  TWh               Asia Pacific  2050  26573.20
1239  TWh                      China  2010   4235.54
1240  TWh                      China  2020   7767.14
1241  TWh                      China  2021   8539.06
1242  TWh                      China  2030  11136.30
1243  TWh                      China  2050  14341.90
1246  TWh                      India  2010    974.48
1247  TWh                      India  2020   1533.31
1248  TWh                      India  2021   1685.98
1249  TWh                      India  2030   2707.78
1250  TWh                      India  2050   5298.35
1253  TWh                      Japan  2010   1164.02
1254  TWh                      Japan  2020   1009.04
1255  TWh                      Japan  2021   1023.64
1256  TWh                      Japan  2030    969.38
1257  TWh                     

In [52]:
# Transpose the columns so year is now the index.
def proj_transpose(df):
    year_index = list(df['Year'].unique())
    newdf = pd.DataFrame(index = year_index)
    for row in df.index:
        col = df.loc[row,'Region'] + ' Demand (' + df.loc[row,'Unit'] + ')'
        newdf.loc[df['Year'],col] = df.loc[row,'Value']
    return newdf

df_proj_stat_yr = proj_transpose(df_proj_stat)
df_proj_ann_yr = proj_transpose(df_proj_ann)
df_proj_stat_yr

World Demand (TWh)  North America Demand (TWh)  \
2010             49844.9                      7815.8   
2020             49844.9                      7815.8   
2021             49844.9                      7815.8   
2030             49844.9                      7815.8   
2050             49844.9                      7815.8   

      United States Demand (TWh)  Central and South America Demand (TWh)  \
2010                     6270.34                                 2591.74   
2020                     6270.34                                 2591.74   
2021                     6270.34                                 2591.74   
2030                     6270.34                                 2591.74   
2050                     6270.34                                 2591.74   

      Brazil Demand (TWh)  Europe Demand (TWh)  European Union Demand (TWh)  \
2010              1173.86              5703.31                      3689.06   
2020              1173.86              5703.31                      3689.06   
2021              1173.86              5703.31                      3689.06   
2030              1173.86              5703.31                      3689.06   
2050              1173.86              5703.31                      3689.06   

      Africa Demand (TWh)  Middle East Demand (TWh)  Eurasia Demand (TWh)  \
2010              2337.29                   2886.16                1937.4   
2020              2337.29                   2886.16                1937.4   
2021              2337.29                   2886.16                1937.4   
2030              2337.29                   2886.16                1937.4   
2050              2337.29                   2886.16                1937.4   

      Russia Demand (TWh)  Asia Pacific Demand (TWh)  China Demand (TWh)  \
2010              1376.48                    26573.2             14341.9   
2020              1376.48                    26573.2             14341.9   
2021              1376.48                    26573.2             14341.9   
2030              1376.48                    26573.2             14341.9   
2050              1376.48                    26573.2             14341.9   

      India Demand (TWh)  Japan Demand (TWh)  Southeast Asia Demand (TWh)  
2010             5298.35               991.5                      3142.94  
2020             5298.35               991.5                      3142.94  
2021             5298.35               991.5                      3142.94  
2030             5298.35               991.5                      3142.94  
2050             5298.35               991.5                      3142.94

In [53]:
# A look at the regions included in this Dataset

df_projected_demand['Region'].value_counts()

World                        481
Middle East                  149
Asia Pacific                 149
North America                149
Africa                       149
Southeast Asia               149
Europe                       149
Central and South America    139
European Union               139
Eurasia                      139
China                        128
India                        128
Russia                       128
Brazil                       128
Japan                        118
United States                118
Japan and Korea               10
East of Suez                  10
Other                         10
Atlantic Basin                10
Non-OPEC                       7
OPEC                           7
Name: Region, dtype: int64

Unfortunately, the full dataset with individual country data is behind a paywall. The IEA's **World Energy Outlook 2022** Report splits the world into seven regions on [Page 499](https://iea.blob.core.windows.net/assets/830fe099-5530-48f2-a7c1-11f35d510983/WorldEnergyOutlook2022.pdf#page=499):

* North America
* Central & South America
* Europe
* Africa
* Middle East
* Eurasia
* Asia Pacific

Below I'll create lists to group the countries together, along with other groups the report defines.

**North America**

In [54]:
n_america = ['CAN', 'MEX', 'USA']

**Central & South America**

In [55]:
cen_s_america = ['ATG', 'ARG', 'ABW', 'BHS', 'BRB', 'BLZ', 'BMU', 'VEN', 'BRA',
                 'VGB', 'CYM', 'CHL', 'COL', 'CRI', 'CUB', 'CUW', 'DMA', 'DOM',
                 'ECU', 'SLV', 'FLK', 'GUF', 'GRD', 'GLP', 'GTM', 'GUY', 'HTI',
                 'HND', 'JAM', 'MTQ', 'MSR', 'NIC', 'PAN', 'PRY', 'PER', 'BOL',
                 'KNA', 'LCA', 'SPM', 'SUR', 'TTO', 'TCA', 'URY']

**Europe**

In [56]:
european_u = ['AUT', 'BEL', 'BGR', 'HRV', 'CYP', 'CZE', 'DNK', 'EST', 'FIN',
              'FRA', 'DEU', 'GRC', 'HUN', 'IRL', 'ITA', 'LVA', 'LTU', 'LUX',
              'MLT', 'NLD', 'POL', 'PRT', 'ROU', 'SVK', 'SVN', 'ESP', 'SWE']

In [57]:
europe = european_u + ['ALB', 'BLR', 'BIH', 'GIB', 'ISL', 'ISR', 'OWID_KOS',
                       'MNE', 'MKD', 'NOR', 'MDA', 'SRB', 'CHE', 'TUR', 'UKR',
                       'GBR']

**Africa**

In [58]:
n_africa = ['DZA', 'EGY', 'LBY', 'MAR', 'TUN']

In [59]:
sub_africa = ['AGO', 'BEN', 'BWA', 'BFA', 'BDI', 'CPV', 'CMR', 'CAF', 'TCD',
              'COM', 'CIV', 'COD', 'DJI', 'GNQ', 'ERI', 'ETH', 'GAB', 'GMB',
              'GHA', 'GIN', 'GNB', 'KEN', 'SWZ', 'LSO', 'LBR', 'MDG', 'MWI',
              'MLI', 'MRT', 'MUS', 'MOZ', 'NAM', 'NER', 'NGA', 'COG', 'REU',
              'RWA', 'STP', 'SEN', 'SYC', 'SLE', 'SOM', 'ZAF', 'SSD', 'SDN',
              'TGO', 'UGA', 'TZA', 'ZMB', 'ZWE']

In [60]:
africa = n_africa + sub_africa

**Middle East**

In [61]:
mid_east = ['BHR', 'IRQ', 'IRN', 'JOR', 'KWT', 'LBN', 'OMN', 'QAT', 'SAU',
            'SYR', 'ARE', 'YEM']

**Eurasia**

In [62]:
caspian = ['ARM', 'AZE', 'GEO', 'KAZ', 'KGZ', 'TJK', 'TKM', 'UZB']

In [63]:
eurasia = caspian + ['RUS']

**Asia Pacific**

In [64]:
se_asia = ['BRN', 'KHM', 'IDN', 'LAO', 'MYS', 'MMR', 'PHL', 'SGP', 'THA','VNM']

In [65]:
asia_pacific = se_asia + ['AFG', 'AUS', 'BGD', 'BTN', 'TWN', 'COK', 'PRK',
                          'FJI', 'PYF', 'IND', 'JPN', 'KIR', 'KOR', 'MAC',
                          'MDV', 'MNG', 'NPL', 'NCL', 'NZL', 'PAK', 'PNG',
                          'CHN', 'WSM', 'SLB', 'LKA', 'TLS', 'VUT']

# China: Includes the (People's Republic of) China and Hong Kong, China.
# I'll create a separate DF for linking with other data.
asia_pacific_hkg = asia_pacific + ['HKG']

**Other Divisions**

In [66]:
iea_countries = ['AUS', 'AUT', 'BEL', 'CAN', 'CZE', 'DNK', 'EST', 'FIN', 'FRA',
                 'DEU', 'GRC', 'HUN', 'IRL', 'ITA', 'JPN', 'KOR', 'LTU', 'LUX',
                 'MEX', 'NLD', 'NZL', 'NOR', 'POL', 'PRT', 'SVK', 'ESP', 'SWE',
                 'CHE', 'TUR', 'GBR', 'USA']

In [67]:
oecd = ['AUS', 'AUT', 'BEL', 'CAN', 'CHL', 'COL', 'CRI', 'CZE', 'DNK', 'EST',
        'FIN', 'FRA', 'DEU', 'GRC', 'HUN', 'ISL', 'IRL', 'ISR', 'ITA', 'JPN',
        'KOR', 'LVA', 'LTU', 'LUX', 'MEX', 'NLD', 'NZL', 'NOR', 'POL', 'PRT',
        'SVK', 'SVN', 'ESP', 'SWE', 'CHE', 'TUR', 'GBR', 'USA']

In [68]:
opec = ['DZA', 'AGO', 'COG', 'GNQ', 'GAB', 'IRN', 'IRQ', 'KWT', 'LBY', 'NGA',
        'SAU', 'ARE', 'VEN']

In [69]:
adv_eco = oecd + ['BGR', 'HRV', 'CYP', 'MLT', 'ROU']

In [70]:
world = (n_america + cen_s_america + europe + africa + mid_east + eurasia + 
         asia_pacific)

world_hkg = world + ['HKG']

In [71]:
dev_asia = asia_pacific
dev_asia.remove('AUS')
dev_asia.remove('JPN')
dev_asia.remove('KOR')
dev_asia.remove('NZL')

dev_asia_hkg = dev_asia + ['HKG']

In [72]:
# emg_dev = All other countries not included in the advanced economies regional
# grouping.
emg_dev = world.copy()
for con in adv_eco:
    emg_dev.remove(con)
    
emg_dev_hkg = emg_dev + ['HKG']

In [73]:
l_america = cen_s_america + ['MEX']

In [74]:
# Non‐OECD: All other countries not included in the OECD regional grouping.

non_oecd = world.copy()
for con in oecd:
    non_oecd.remove(con)

non_oecd_hkg = non_oecd + ['HKG']

In [75]:
# Non‐OPEC: All other countries not included in the OPEC regional grouping.

non_opec = world.copy()
for con in opec:
    non_opec.remove(con)
    
non_opec_hkg = non_opec + ['HKG']

Countries that are left out:
Because I don't have corresponding information in other datasets, I had to leave out Palau (PW), Bonaire BQ-BO, Saba, Sint Eustatius (BQ-SE),  Sint Maarten (SX), Saint Vincent and the Grenadines (VC), & Anguilla (AI)  from this dataset. Saint P has a population of about 100,000, and the other  islands combined represent less than 65,000 people.

### Visualize the quantities of the data.

In [ ]:
# This is where I'll build those scary graphs that show the hockey stick curves
# and such.

## Preparing Data For Analysis

### Testing for bad data

## Frequency Distributions

# Building Models
It's finally time to build the classification models. I will fit each vectorization technique with the following models:
1. Multinomial Naive Bayes 
2. Logistic Regression
3. Support Vector Machine
4. RandomForest
5. XGBoost

I'll explain each of the models below in the same section in which they are created. Before I start, here's one last preview of the DataFrame I am about to model.

In [ ]:
# Time-Series Concentration of CO2 and NH4 based on emissions of the same.


## Final Model Selection

# Tuning Models

# Final Model And Analysis

# Conclusion

## Limitations of Scope / Provided Data

## Future Research Opportunities

## Recommendations

That's a wrap! Thanks for hiring Greg Osborne and I hope to work with you more in the future.

## Grading

***To pass this project, you must pass each rubric objective.*** The project rubric objectives for Capstone are:

1. Attention to Detail
2. Project Motivation
3. Independent Learning
4. Reproducibility

### Attention to Detail

Once again, the Attention to Detail standard has increased. ***In Capstone, you need to complete 100% (10 out of 10) or more of the checklist elements in order to pass the Attention to Detail objective.***

**NOTE THAT THE PASSING BAR IS HIGHER IN CAPSTONE THAN IT WAS IN PHASE 4!**

#### Exceeds Objective

In addition to completing 100% of the checklist items, goes above and beyond to create a compelling, accessible repository

> Some things to consider doing are running a spell-checker to ensure you don't have typos, adding a description to the repository, using Markdown extensively in the README to create a skim-able document, etc.

> You can also look at the [GitHub tips for post-graduation](https://docs.google.com/document/d/1cT13597jlbiVgKUFyhGC7_QFUrLQUfZViRu8kD_ErAA/edit?usp=sharing) doc (which your career coach will be sending you soon!) to find more examples of enhancements

#### Meets Objective (Passing Bar)

100% of the project checklist items are complete


### Project Motivation

The passing bar for this element will be communicated in the **elevator pitch** portion of the README, immediately after the project title. The elevator pitch should:

* Describe the **data** used in the model
   * This does *not* mean that you should list out every single column name and data type. Instead, describe the data generally.
* Identify the **target**
   * What are you predicting?
* Communicate the **rationale** for predicting this target with this data
   * This can be very brief. The goal is to avoid a "so what?" response
* Evaluate the final model's **performance**

Here are some examples of elevator pitches, which include all of the required elements:

> Searching for skincare products can be overwhelming, since there are so many options and skincare needs vary so much from person to person.  This project uses product and review data scraped from SkinStore.com to make personalized recommendations to a new user based on their skin type, skin problems, and the types of products they're looking for. Using the `surprise` Python package, the final machine learning model has an RMSE of 0.96, meaning that it is able to predict the star rating (out of 5) within 1 star for a given product and user.

> Taxi drivers in New York City have the flexibility to choose when and where to start their workdays, but the best time and location varies. This project uses data about taxi trips from the NYC Open Data portal as well as weather and CitiBike data to forecast demand for taxi pickups across different neighborhoods. The model is able to explain about 70% of the variance in demand and would be especially useful for a taxi company to orchestrate their distribution of resources.

**Note:** There is NO threshold for model performance you are required to meet. So long as you explain what you tried and properly evaluate the outcome, it's fine to have a model with poor performance. We recognize that your time is limited during Capstone and we want you to focus more on the process and communication, not worrying about model performance.

#### Exceeds Objective

Weaves a clear narrative throughout the project explaining the scope, methods, and specific use cases for the final model

#### Meets Objective (Passing Bar)

Describes data, identifies a target, and communicates the rationale for predicting the target with this data and an evaluation of the final model's performance

### Independent Learning

Learning the specific content we teach in the Flatiron School Data Science program is important, but learning how to learn is even more important! This objective is asking you to flex that skill.

The baseline requirement is that you do some domain research, rather than just diving into the dataset. Sometimes a clear understanding of the domain will help you collect a key feature or perform a key preprocessing step that will massively improve your model performance. Work smarter, not harder!

#### Exceeds Objective

Demonstrates a "deep dive" beyond the course content of the DS program, such as extensive domain knowledge or extensively using additional Python packages

> Ideally, your notebook would teach your *instructor* something new about a domain or a technique!

#### Meets Objective (Passing Bar)

Cites at least one external source to demonstrate domain understanding of the project topic

> The source is up to you. It can be an academic whitepaper, an industry publication, a newspaper article, or even just the Wikipedia page on a topic.


### Reproducibility

While you may explore additional forms of *deployment* such as creating an API or web app, the main way that Flatiron School DS projects are deployed is through a reproducible notebook on GitHub.

The explanation of how to reproduce your analysis can either be directly in your README in the Repository Navigation section, or you can create a separate Markdown file and link it in the README.

Be sure to describe the operating system used (including cloud systems such as Kaggle or Google Colab), the packages, and how to get the data.

#### Exceeds Objective

Writes a script so that reproducing the project is seamless on a designated platform

> This can be a Python script or a terminal script. It might involve downloading images and moving them into subfolders, installing packages with `conda` or `pip`, etc.

#### Meets Objective (Passing Bar)

Incorporates the basic elements of a reproducible project: a description of software used + how to get the data

> Ideally you would list the specific versions of packages used through an `environment.yml` or `requirements.txt` file. If you are using a platform like Kaggle, describe the additional `pip install` commands required beyond the base Kaggle environment.

> If your data is proprietary or otherwise not accessible to someone trying to reproduce your analysis, make sure you explain this clearly. If possible, include a sample or anonymized version of the data in the repo.


### Legacy Data Sources
These data sources represent content I found but did not use.

#### Disaster Stats
* International Disaster Database: [EM-Dat](https://www.emdat.be/), [Column Definitions](https://public.emdat.be/about)

#### Refugee Stats
* All the refugee info you could ever want, split any way you want: [UNHCR](https://www.unhcr.org/refugee-statistics/download/)
* More displacement data on refugees [Internal-Displacement](https://www.internal-displacement.org/database/displacement-data)
* A mega dive into the reasoning behind the shocking estimates of future climate refugees: [Migration Data Portal](https://www.migrationdataportal.org/themes/environmental_migration_and_statistics#data-sources)

#### Population Stats
* Population from 1500–2000: [Our World in Data](https://ourworldindata.org/grapher/population-by-country)
* Population from 1950–2021 and future projections: [UN's Website](https://population.un.org/wpp/Download/Standard/MostUsed/), [Dataset](https://population.un.org/wpp/Download/Files/1_Indicators%20(Standard)/EXCEL_FILES/1_General/WPP2022_GEN_F01_DEMOGRAPHIC_INDICATORS_COMPACT_REV1.xlsx)

#### GDP Stats
* GDP from 1820–2018 [Our World in Data](https://ourworldindata.org/grapher/gdp-world-regions-stacked-area)
* GDP from 1960–2021 in 2015 USD [The World Bank](https://data.worldbank.org/indicator/NY.GDP.MKTP.KD)
* GDP growth rates for future years to project GDP past 2021 [International Monetary Fund](https://www.imf.org/external/datamapper/NGDP_RPCH@WEO/OEMDC/ADVEC/WEOWORLD)

#### At-Risk Property Stats
* The total cost, per country, of buildings and machinery at risk of destruction, Produced Capital [CWON 2021 Country Tool (Full Dataset - 219 countries)](https://datacatalog.worldbank.org/search/dataset/0042066)

#### Food Production
* Crop yields under three different climate scenarios: [NASA via Data World](https://data.world/nasa/potential-impacts-of-climate-change-on-world-food-supply-v-1/workspace/file?filename=potential-impacts-world-food-supply.xls)
* Another dataset regarding crop yields: [Kaggle](https://www.kaggle.com/datasets/thedevastator/the-relationship-between-crop-production-and-cli)

#### The 2030 target, keeping with the Paris Agreement
* The Paris Agreement is a legally binding document signed by 186 countries intended to limit global temperature rise to 1.5 °C: [The UN website](https://unfccc.int/process-and-meetings/the-paris-agreement)
  * "The Paris Agreement is a legally binding international treaty on climate change. It was adopted by 196 Parties at the UN Climate Change Conference (COP21) in Paris, France, on 12 December 2015. It entered into force on 4 November 2016.
  
    "Its overarching goal is to hold “the increase in the global average temperature to well below 2°C above pre-industrial levels” and pursue efforts “to limit the temperature increase to 1.5°C above pre-industrial levels.”"
    
* In 2018, the gauntlet was thrown, to cut emissions by 45% from 2010 levels by 2030: [IPCC 2018 Special Report – Global Warming of 1.5 ºC](https://www.ipcc.ch/sr15/chapter/spm/#article-spm-c)
  * "C.1. In model pathways with no or limited overshoot of 1.5°C, global net anthropogenic CO2 emissions decline by about 45% from 2010 levels by 2030 (40–60% interquartile range), reaching net zero around 2050 (2045–2055 interquartile range)."

# Possible Project Directions:

## Show Potential Donors That The Financial Cost of Disaster Mitigation Efforts Will Continue To Rise
### Stakeholder: FEMA? UN Disaster Org? Red Cross?
Plot out all of the climate related disasters of the last 120 years and stress that the pattern and scale are escalating.

## Illustrate Climate Change's Future Effects on the Global Food Supply
### Stakeholder: Hunger NGO? Compassion International? Food For The Hungry?
Show that the already precarious state of food insecurity will reach frightening levels in the near future. Show data along with population increase data.

## Refugee projection for the United States
### Stakeholder: [The Subcommittee on Immigration Integrity, Security, and Enforcement](https://judiciary.house.gov/subcommittees/committee-judiciary/subcommittee-immigration-integrity-security-and-enforcement)
This would be a mix of the disaster statistics and refugee numbers in general. The deeper I dove into this, the more complicated it got. Eventually I found that the IPCC itself admits that the estimating this has been very difficult. If I did this, it would be a stab in the dark based on just the data I listed in the above cell. Mission creep would be extremely tempting with this one.

The SSP quantifications and the data in this database should be cited as:
-------------------------------------------------------------------------

Download citations: [bibtex (.bib)](https://secure.iiasa.ac.at/web-apps/ene/SspDb/static/bib/SSP__v2_RecommCit.bib) | [Reference manager (.ris)](https://secure.iiasa.ac.at/web-apps/ene/SspDb/static/bib/SSP__v2_RecommCit.ris)

Keywan Riahi, Detlef P. van Vuuren, Elmar Kriegler, Jae Edmonds, Brian C. O'Neill, Shinichiro Fujimori, Nico Bauer, Katherine Calvin, Rob Dellink, Oliver Fricko, Wolfgang Lutz, Alexander Popp, Jesus Crespo Cuaresma, Samir KC, Marian Leimbach, Leiwen Jiang, Tom Kram, Shilpa Rao, Johannes Emmerling, Kristie Ebi, Tomoko Hasegawa, Petr Havlík, Florian Humpenöder, Lara Aleluia Da Silva, Steve Smith, Elke Stehfest, Valentina Bosetti, Jiyong Eom, David Gernaat, Toshihiko Masui, Joeri Rogelj, Jessica Strefler, Laurent Drouet, Volker Krey, Gunnar Luderer, Mathijs Harmsen, Kiyoshi Takahashi, Lavinia Baumstark, Jonathan C. Doelman, Mikiko Kainuma, Zbigniew Klimont, Giacomo Marangoni, Hermann Lotze-Campen, Michael Obersteiner, Andrzej Tabeau, Massimo Tavoni.\
**The Shared Socioeconomic Pathways and their energy, land use, and greenhouse gas emissions implications: An overview**, Global Environmental Change, Volume 42, Pages 153-168, 2017,\
ISSN 0959-3780, DOI:[110.1016/j.gloenvcha.2016.05.009](https://doi.org/10.1016/j.gloenvcha.2016.05.009)

### In addition, please cite also the following papers if you are using

**1) the data of the IAM scenarios on low stabilization scenarios for 1.9 W/m2, or\
2) the harmonized CMIP6 emissions data set:**

1.  Rogelj, J., Popp, A., Calvin, K.V., Luderer, G., Emmerling, J., Gernaat, D., Fujimori, S., Strefler, J., Hasegawa, T., Marangoni, G., Krey, V., Kriegler, E., Riahi, K., van Vuuren, D.P., Doelman, J., Drouet, L., Edmonds, J., Fricko, O., Harmsen, M., Havlik, P., Humpenöder, F., Stehfest, E., Tavoni, M., **Scenarios towards limiting global mean temperature increase below 1.5 °C**. Nature Climate Change 8, 2018, 325-332.\
    DOI:[10.1038/s41558-018-0091-3](https://doi.org/10.1038/s41558-018-0091-3)
2.  Gidden, M. J., Riahi, K., Smith, S. J., Fujimori, S., Luderer, G., Kriegler, E., van Vuuren, D. P., van den Berg, M., Feng, L., Klein, D., Calvin, K., Doelman, J. C., Frank, S., Fricko, O., Harmsen, M., Hasegawa, T., Havlik, P., Hilaire, J., Hoesly, R., Horing, J., Popp, A., Stehfest, E., and Takahashi, K.: **Global emissions pathways under different socioeconomic scenarios for use in CMIP6: a dataset of harmonized emissions trajectories through the end of the century**, Geosci. Model Dev. Dev., 12, 1443-1475,\
    DOI:[10.5194/gmd-12-1443-2019](https://doi.org/10.5194/gmd-12-1443-2019), 2019.

### For the five SSP IAM "marker" scenario sets please use the following citations:

Download citations: [bibtex (.bib)](https://secure.iiasa.ac.at/web-apps/ene/SspDb/static/bib/SSP__v2_Marker.bib) | [Reference manager (.ris)](https://secure.iiasa.ac.at/web-apps/ene/SspDb/static/bib/SSP__v2_Marker.ris)

**SSP1**: Detlef P. van Vuuren, Elke Stehfest, David E.H.J. Gernaat, Jonathan C. Doelman, Maarten van den Berg, Mathijs Harmsen, Harmen Sytze de Boer, Lex F. Bouwman, Vassilis Daioglou, Oreane Y. Edelenbosch, Bastien Girod, Tom Kram, Luis Lassaletta, Paul L. Lucas, Hans van Meijl, Christoph Müller, Bas J. van Ruijven, Sietske van der Sluis, Andrzej Tabeau, **Energy, land-use and greenhouse gas emissions trajectories under a green growth paradigm**, Global Environmental Change, Volume 42, 2017, Pages 237-250,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2016.05.008](https://doi.org/10.1016/j.gloenvcha.2016.05.008)

**SSP2:** Oliver Fricko, Petr Havlik, Joeri Rogelj, Zbigniew Klimont, Mykola Gusti, Nils Johnson, Peter Kolp, Manfred Strubegger, Hugo Valin, Markus Amann, Tatiana Ermolieva, Nicklas Forsell, Mario Herrero, Chris Heyes, Georg Kindermann, Volker Krey, David L. McCollum, Michael Obersteiner, Shonali Pachauri, Shilpa Rao, Erwin Schmid, Wolfgang Schoepp, Keywan Riahi, **The marker quantification of the Shared Socioeconomic Pathway 2: A middle-of-the-road scenario for the 21st century**, Global Environmental Change, Volume 42, 2017, Pages 251-267,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2016.06.004](https://doi.org/10.1016/j.gloenvcha.2016.06.004)

**SSP3:** Shinichiro Fujimori, Tomoko Hasegawa, Toshihiko Masui, Kiyoshi Takahashi, Diego Silva Herran, Hancheng Dai, Yasuaki Hijioka, Mikiko Kainuma, **SSP3: AIM implementation of Shared Socioeconomic Pathways**, Global Environmental Change, Volume 42, 2017, Pages 268-283,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2016.06.009](https://doi.org/10.1016/j.gloenvcha.2016.06.009)

**SSP4:** Katherine Calvin, Ben Bond-Lamberty, Leon Clarke, James Edmonds, Jiyong Eom, Corinne Hartin, Sonny Kim, Page Kyle, Robert Link, Richard Moss, Haewon McJeon, Pralit Patel, Steve Smith, Stephanie Waldhoff, Marshall Wise, **The SSP4: A world of deepening inequality**, Global Environmental Change, Volume 42, 2017, Pages 284-296,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2016.06.010](https://doi.org/10.1016/j.gloenvcha.2016.06.010)

**SSP5:** Elmar Kriegler, Nico Bauer, Alexander Popp, Florian Humpenöder, Marian Leimbach, Jessica Strefler, Lavinia Baumstark, Benjamin Leon Bodirsky, Jérôme Hilaire, David Klein, Ioanna Mouratiadou, Isabelle Weindl, Christoph Bertram, Jan-Philipp Dietrich, Gunnar Luderer, Michaja Pehl, Robert Pietzcker, Franziska Piontek, Hermann Lotze-Campen, Anne Biewald, Markus Bonsch, Anastasis Giannousakis, Ulrich Kreidenweis, Christoph Müller, Susanne Rolinski, Anselm Schultes, Jana Schwanitz, Miodrag Stevanovic, Katherine Calvin, Johannes Emmerling, Shinichiro Fujimori, Ottmar Edenhofer, **Fossil-fueled development (SSP5): An energy and resource intensive scenario for the 21st century**, Global Environmental Change, Volume 42, 2017, Pages 297-315,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2016.05.015](https://doi.org/10.1016/j.gloenvcha.2016.05.015)

### For the basic elements of the SSPs please cite in addition to the SSP overview also the following papers:

Download citations: [bibtex (.bib)](https://secure.iiasa.ac.at/web-apps/ene/SspDb/static/bib/SSP__v2_Basic.bib) | [Reference manager (.ris)](https://secure.iiasa.ac.at/web-apps/ene/SspDb/static/bib/SSP__v2_Basic.ris)

**Population:** Samir KC, Wolfgang Lutz, **The human core of the shared socioeconomic pathways: Population scenarios by age, sex and level of education for all countries to 2100**, Global Environmental Change, Volume 42, 2017, Pages 181-192,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2014.06.004](https://doi.org/10.1016/j.gloenvcha.2014.06.004)

**GDP marker projections by the OECD:** Rob Dellink, Jean Chateau, Elisa Lanzi, Bertrand Magné, **Long-term economic growth projections in the Shared Socioeconomic Pathways**, Global Environmental Change, Volume 42, 2017, Pages 200-214,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2015.06.004](https://doi.org/10.1016/j.gloenvcha.2015.06.004)

**GDP projections by IIASA:** Jesús Crespo Cuaresma, **Income projections for climate change research: A framework based on human capital dynamics**, Global Environmental Change, Volume 42, 2017, Pages 226-236,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2015.02.012](https://doi.org/10.1016/j.gloenvcha.2015.02.012)

**GDP Projections by PIK:** Marian Leimbach, Elmar Kriegler, Niklas Roming, Jana Schwanitz, **Future growth patterns of world regions -- A GDP scenario approach**, Global Environmental Change, Volume 42, 2017, Pages 215-225,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2015.02.005](https://doi.org/10.1016/j.gloenvcha.2015.02.005)

**Urbanization:** Leiwen Jiang, Brian C. O'Neill, **Global urbanization projections for the Shared Socioeconomic Pathways**, Global Environmental Change, Volume 42, 2017, Pages 193-199,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2015.03.008](https://doi.org/10.1016/j.gloenvcha.2015.03.008)

### Additional detail about the energy, Land-use and air pollution trends of the SSP IAM scenarios can be found here:

Download citations: [bibtex (.bib)](https://secure.iiasa.ac.at/web-apps/ene/SspDb/static/bib/SSP__v2_AddDetail.bib) | [Reference manager (.ris)](https://secure.iiasa.ac.at/web-apps/ene/SspDb/static/bib/SSP__v2_AddDetail.ris)

**Land-use:** Alexander Popp, Katherine Calvin, Shinichiro Fujimori, Petr Havlik, Florian Humpenöder, Elke Stehfest, Benjamin Leon Bodirsky, Jan Philipp Dietrich, Jonathan C. Doelmann, Mykola Gusti, Tomoko Hasegawa, Page Kyle, Michael Obersteiner, Andrzej Tabeau, Kiyoshi Takahashi, Hugo Valin, Stephanie Waldhoff, Isabelle Weindl, Marshall Wise, Elmar Kriegler, Hermann Lotze-Campen, Oliver Fricko, Keywan Riahi, Detlef P. van Vuuren, **Land-use futures in the shared socio-economic pathways**, Global Environmental Change, Volume 42, 2017, Pages 331-345,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2016.2016.10.002](https://doi.org/10.1016/j.gloenvcha.2016.10.002)

**Air pollution:** Shilpa Rao, Zbigniew Klimont, Steven J. Smith, Rita Van Dingenen, Frank Dentener, Lex Bouwman, Keywan Riahi, Markus Amann, Benjamin Leon Bodirsky, Detlef P. van Vuuren, Lara Aleluia Reis, Katherine Calvin, Laurent Drouet, Oliver Fricko, Shinichiro Fujimori, David Gernaat, Petr Havlik, Mathijs Harmsen, Tomoko Hasegawa, Chris Heyes, Jérôme Hilaire, Gunnar Luderer, Toshihiko Masui, Elke Stehfest, Jessica Strefler, Sietske van der Sluis, Massimo Tavoni. **Future air pollution in the Shared Socio-economic Pathways**, Global Environmental Change, Volume 42, 2017, Pages 346-358,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2016.2016.05.012](https://doi.org/10.1016/j.gloenvcha.2016.05.012)

**Energy:** Nico Bauer, Katherine Calvin, Johannes Emmerling, Oliver Fricko, Shinichiro Fujimori, Jérôme Hilaire, Jiyong Eom, Volker Krey, Elmar Kriegler, Ioanna Mouratiadou, Harmen Sytze de Boer, Maarten van den Berg, Samuel Carrara, Vassilis Daioglou, Laurent Drouet, James E. Edmonds, David Gernaat, Petr Havlik, Nils Johnson, David Klein, Page Kyle, Giacomo Marangoni, Toshihiko Masui, Robert C. Pietzcker, Manfred Strubegger, Marshall Wise, Keywan Riahi, Detlef P. van Vuuren, **Shared Socio-Economic Pathways of the Energy Sector -- Quantifying the Narratives**, Global Environmental Change, Volume 42, 2017, Pages 316-330,\
ISSN 0959-3780, DOI:[10.1016/j.gloenvcha.2016.07.006](https://doi.org/10.1016/j.gloenvcha.2016.07.006)

### Historical Emissions Inventories:

Download citations: [bibtex (.bib)](https://secure.iiasa.ac.at/web-apps/ene/SspDb/static/bib/SSP__v2_Inventories.bib) | [Reference manager (.ris)](https://secure.iiasa.ac.at/web-apps/ene/SspDb/static/bib/SSP__v2_Inventories.ris)

Velders, G.J.M., D.W. Fahey, J.S. Daniel, S.O. Andersen, M. McFarland, **Future atmospheric abundances and climate forcings from scenarios of global and regional hydrofluorocarbon (HFC) emissions**, Atmospheric Environment, 123, 200-209, 2015.\
DOI:[10.1016/j.atmosenv.2015.10.071](https://doi.org/10.1016/j.atmosenv.2015.10.071)

van Marle, M. J. E., Kloster, S., Magi, B. I., Marlon, J. R., Daniau, A.-L., Field, R. D., Arneth, A., Forrest, M., Hantson, S., Kehrwald, N. M., Knorr, W., Lasslop, G., Li, F., Mangeon, S., Yue, C., Kaiser, J. W., and van der Werf, G. R.: **Historic global biomass burning emissions based on merging satellite observations with proxies and fire models (1750-2015)**, Geoscientific Model Development Discussions, 2017\
DOI:[10.5194/gmd-10-3329-2017](https://doi.org/10.5194/gmd-10-3329-2017)

Hoesly, R. M., Smith, S. J., Feng, L., Klimont, Z., Janssens-Maenhout, G., Pitkanen, T., Seibert, J. J., Vu, L., Andres, R. J., Bolt, R. M., Bond, T. C., Dawidowski, L., Kholod, N., Kurokawa, J.-I., Li, M., Liu, L., Lu, Z., Moura, M. C. P., O'Rourke, P. R., and Zhang, Q.: **Historical (1750-2014) anthropogenic emissions of reactive gases and aerosols from the Community Emissions Data System (CEDS)**, Geoscientific Model Development, 11, 369- 408, 2018.\
DOI:[10.5194/gmd-11-369-2018](https://doi.org/10.5194/gmd-11-369-2018)

Gütschow, J., Jeffery, M. L., Gieseke, R., Gebel, R., Stevens, D., Krapp, M., and Rocha, M.: **The PRIMAP-hist national historical emissions time series**, Earth System Science Data, 8, 571-603, 2016\
DOI:[10.5194/essd-8-571-2016](https://doi.org/10.5194/essd-8-571-2016)

L.J. Carpenter and S. Reimann (Lead Authors), J.B. Burkholder, C. Clerbaux, B.D. Hall, R. Hossaini, J.C. Laube, and S.A. Yvon-Lewis, Ozone-Depleting Substances (ODSs) and Other Gases of Interest to the Montreal Protocol, Chapter 1 in **Scientific Assessment of Ozone Depletion: 2014**, Global Ozone Research and Monitoring Project - Report No. 55, World Meteorological Organization, Geneva, Switzerland, 2014\
ISBN: 978-9966-076-01-4, <http://www.wmo.int/pages/prog/arep/gaw/ozone_2014/documents/3_Chapter1_2014OzoneAssessment.pdf>